File: worsham.twitter.clone.angular/Models/TweetAndRelatedCommentsViewModel.cs
using worsham.twitter.clone.angular.Models.EntityModels;

namespace worsham.twitter.clone.angular.Models
{
    public class TweetAndRelatedCommentsViewModel
    {
        // public Tweets? Tweet { get; set; }
        public string? TweetOwnerName { get; set; }
        public string? TweetOwnersProfilePicture { get; set; }

        public string? TweetContent { get; set; }
        public DateTime TweetCreationDateTime { get; set; }
        public List<CommentModelView>? TweetComments { get; set; }
    }

    public class CommentModelView
    {
        public int CommentId { get; set; }
        public int CommenterId { get; set; }
        public string? CommenterUserName { get; set; }
        public string? CommentersProfilePicture { get; set; }
        public string? CommentContent { get; set; }
    }
}

File: worsham.twitter.clone.angular/ClientApp/src/app/models/tweetAndRelatedCommentsViewModel.ts
export class TweetAndRelatedCommentsViewModel {
  constructor(
    public tweetOwnerName: string,
    public tweetOwnersProfilePicture: string,
    public tweetContent: string,
    public tweetCreationDateTime: string,
    public tweetComments: CommentModelView[]
    ) {}
}

export class CommentModelView {
  constructor(
    public commentId: number,
    public commenterId: number,
    public commenterUserName: string,
    public commentersProfilePicture: string,
    public commentContent: string
  ){}
}
